core/trie/src/lib.rs

```diff
@@ -317,6 +315,138 @@ fn branch_node(has_value: bool, has_children: impl Iterator<Item = bool>) -> [u8
 	[first, (bitmap % 256 ) as u8, (bitmap / 256 ) as u8]
 }
 
+// TODO EMCH issue to add default value to HashDB trait: avoiding this costy calculation here
```

Comment:
KeyedSpaceDB way of storing null hash is not right: I think we should add access to this info in hashdb trait and keyvaluedb trait. But this needs changing hashdb trait which is beyond this PR immediate scope, but an issue can be created.

Reply:
resolved as an issue was created for it